include/dump_offload.hpp

```diff
@@ -0,0 +1,519 @@
+                }
+                return;
+            }
+            BMCWEB_LOG_CRITICAL << "INFO: Reset OffloadUri of " << dumpType
+                                << " dump id " << entryID;
+            },
+            "xyz.openbmc_project.Dump.Manager",
+            "/xyz/openbmc_project/dump/" + dumpType + "/entry/" + entryID,
+            "org.freedesktop.DBus.Properties", "Set",
+            "xyz.openbmc_project.Dump.Entry", "OffloadUri",
+            std::variant<std::string>(value));
+    }
+
+    void cleanupSocketFiles()
+    {
+        std::error_code ec;
+        bool fileExists = std::filesystem::exists(unixSocketPath, ec);
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "Unix socket file clean up failed "
+                             << unixSocketPath;
+            this->connection->sendStreamErrorStatus(
+                boost::beast::http::status::internal_server_error);
+            return;
+        }
+        if (fileExists)
+        {
+            unixSocket.close();
+            std::remove(unixSocketPath.c_str());
+        }
+        return;
+    }
+
+    void getDumpSize(const std::string& entryID, const std::string& dumpType)
+    {
+        crow::connections::systemBus->async_method_call(
+            [this,
+             self(shared_from_this())](const boost::system::error_code ec,
+                                       const std::variant<uint64_t>& size) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR
+                    << "DBUS response error: Unable to get the dump size "
+                    << ec;
+
+                if (ec.value() == EBADR)
+                {
+                    this->connection->sendStreamErrorStatus(
+                        boost::beast::http::status::not_found);
+                }
+                else
+                {
+                    this->connection->sendStreamErrorStatus(
+                        boost::beast::http::status::internal_server_error);
+                }
+                this->connection->close();
+                this->cleanupSocketFiles();
+                return;
+            }
+            const uint64_t* dumpsize = std::get_if<uint64_t>(&size);
+            if (dumpsize == nullptr)
+            {
+                BMCWEB_LOG_ERROR << "DBUS response error: Unable to get "
+                                    "the dump size value"
+                                 << ec;
+                this->connection->sendStreamErrorStatus(
+                    boost::beast::http::status::internal_server_error);
+                this->connection->close();
+                this->cleanupSocketFiles();
+                return;
+            }
+            this->dumpSize = *dumpsize;
+            this->initiateOffload();
+            this->doConnect();
+            },
+            "xyz.openbmc_project.Dump.Manager",
+            "/xyz/openbmc_project/dump/" + dumpType + "/entry/" + entryID,
+            "org.freedesktop.DBus.Properties", "Get",
```

Comment:
The new code all uses    sdbusplus::asio::getProperty
https://github.com/ibm-openbmc/bmcweb/blob/1050/redfish-core/lib/systems.hpp#L208